test/Sign.Core.Test/SignatureProviders/AppInstallerServiceSignatureProviderTests.cs

```diff
@@ -0,0 +1,58 @@
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace Sign.Core.Test
+{
+    public class AppInstallerServiceSignatureProviderTests
+    {
+        private readonly AppInstallerServiceSignatureProvider _provider;
+
+        public AppInstallerServiceSignatureProviderTests()
+        {
+            _provider = new AppInstallerServiceSignatureProvider(
+                Mock.Of<IKeyVaultService>(),
+                Mock.Of<ILogger<ISignatureProvider>>());
+        }
+
+        [Fact]
+        public void Constructor_WhenKeyVaultServiceIsNull_Throws()
+        {
+            ArgumentNullException exception = Assert.Throws<ArgumentNullException>(
+                () => new AppInstallerServiceSignatureProvider(
+                    keyVaultService: null!,
+                    Mock.Of<ILogger<ISignatureProvider>>()));
+
+            Assert.Equal("keyVaultService", exception.ParamName);
+        }
+
+
+        [Fact]
+        public void Constructor_WhenLoggerIsNull_Throws()
+        {
+            ArgumentNullException exception = Assert.Throws<ArgumentNullException>(
+                () => new AppInstallerServiceSignatureProvider(
+                    Mock.Of<IKeyVaultService>(),
+                    logger: null!));
+
+            Assert.Equal("logger", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData(".appinstaller")]
```

Comment:
Is culture testing like turkish i interesting here (case insensitivity has varying behaviors across culture setting)? Or is it not worth the effort?

Reply:
What code do we have that's culture-specific? Our inputs/outputs should all be invariant as they're filenames? If anything, we might want to add the analyzer that checks for specifying the CultureInfo and ensure it's always Invariant?

Reply:
I don't know the full answer to your question :) I asked because I saw a pattern I'm familiar with. Case insensitivity works differently on different cultures. Example: https://dotnetfiddle.net/a3vQK1

I dug into the code and we use the invariant so it should not be culture dependent. 
https://github.com/dotnet/sign/blob/6d16cdda0c06336041ce283fbca65d3aad1683cc/src/Sign.Core/SignatureProviders/AppInstallerServiceSignatureProvider.cs#LL30C13-L30C111

So the product code looks fine, but (theoretically) the test code could have a test for `.appİnstaller` and `.appınstaller` to ensure that `CanSign` returns false. It's also fine to say "that test is too crazy to write" 😄.

The analyzer idea is great. I think there are some global readiness analyzers that NuGet Client team has used recently.

Reply:
I pushed a commit that:

1.  Enabled analyzers for several string comparison issues.  Notably, I did not resolve CA1305 warnings or enable CA1305 analyzer warnings; a work item tracking that is https://github.com/dotnet/sign/issues/450.
2.  Changed file name comparison from `InvariantCultureIgnoreCase` to `OrdinalIgnoreCase`, which is consistent with guidance [here](https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings?WT.mc_id=DT-MVP-5003978#choosing-a-stringcomparison-member-for-your-method-call).
3.  Added unit tests, especially for Turkish i.